GoRogue/MapGeneration/Region.cs

```diff
@@ -0,0 +1,367 @@
+        public void AddConnection(Point connection)
+        {
+            Remove(connection);
+            _connections.Add(connection);
+        }
+
+        /// <summary>
+        /// Removes (from this region) the outer points shared in common
+        /// with the imposing region
+        /// </summary>
+        /// <param name="imposing">the region to check for common outer points</param>
+        public void RemoveOverlappingOuterPoints(Region imposing)
+        {
+            foreach (Point c in imposing.OuterPoints)
+            {
+                while (_outerPoints.Contains(c))
+                    _outerPoints.Remove(c);
+                while (_innerPoints.Contains(c))
+                    _innerPoints.Remove(c);
+            }
+        }
+
+        /// <summary>
+        /// Removes (from this region) the inner points shared in common
+        /// with the imposing region
+        /// </summary>
+        /// <param name="imposing">the region to check for common outer points</param>
+        public void RemoveOverlappingInnerPoints(Region imposing)
+        {
+            foreach (Point c in imposing.InnerPoints)
+            {
+                while (_outerPoints.Contains(c))
+                    _outerPoints.Remove(c);
+                while (_innerPoints.Contains(c))
+                    _innerPoints.Remove(c);
+            }
+        }
+        /// <summary>
+        /// Removes (from this region) the outer points shared in common
+        /// with the imposing region
+        /// </summary>
+        /// <param name="imposing">the region to check for common outer points</param>
+        public void RemoveOverlappingPoints(Region imposing)
+        {
+            foreach (Point c in imposing.Points)
+            {
+                while (_outerPoints.Contains(c))
+                    _outerPoints.Remove(c);
+                while (_innerPoints.Contains(c))
+                    _innerPoints.Remove(c);
+            }
+        }
+
+        /// <summary>
+        /// Rotates this region by an arbitrary number of degrees
+        /// </summary>
+        /// <param name="degrees">The amount of degrees to rotate this region</param>
+        /// <param name="doToSelf">Whether to perform this rotation on this region, or return a new region</param>
+        /// <param name="origin">The Point around which to rotate</param>
+        /// <returns>This region, rotated</returns>
+        /// <remarks>
+        /// This is destructive to the region's Connections, so you should try to refrain from generating those
+        /// until after you've performed your rotations.
+        /// </remarks>
+        public virtual Region Rotate(float degrees, bool doToSelf, Point origin = default)
+        {
+            int quarterTurns = 0;
+            double radians = SadRogue.Primitives.MathHelpers.ToRadian(degrees);
+
+            List<Point> corners = new List<Point>();
+            if (origin == default(Point))
```

Comment:
`Point.None`.  Due to the way structs work, default is all-0's, so (0, 0), which very well could be a valid origin